pkg/collector/corechecks/servicediscovery/discover_services_linux.go

```diff
@@ -0,0 +1,318 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package servicediscovery
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/prometheus/procfs"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/servicedetector"
+	"github.com/DataDog/datadog-agent/pkg/process/portlist"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func init() {
+	newOSImpl = newLinuxImpl
+}
+
+type aliveMap struct {
```

Comment:
1. why do these maps need to be concurrency-safe?
2. if we need them, we should use a generic thread-safe type, since the logic and key are identical; all that's different is the data stored in the map.

Reply:
I was previously spawning goroutines, but I'm not doing this anymore so I removed this code.